code/modules/mob/living/autohiss.dm

```diff
@@ -10,7 +10,7 @@
 	return message // no autohiss at this level
 
 /mob/living/carbon/human/handle_autohiss(message, datum/language/L)
-	if(!client || client.autohiss_mode == AUTOHISS_OFF) // no need to process if there's no client or they have autohiss off
+	if(!client || client.prefs.autohiss_mode == AUTOHISS_OFF) // no need to process if there's no client or they have autohiss off
 		return message
 	return species.handle_autohiss(message, L, client.autohiss_mode)
```

Comment:
You didn't get rid of client.autohiss_mode, so the preference doesn't even work correctly